schematic/visualization/attributes_explorer.py

```diff
@@ -22,34 +23,44 @@ class AttributesExplorer:
     def __init__(
         self,
         path_to_jsonld: str,
-        data_model_labels: DisplayLabelType,
+        data_model_labels: DisplayLabelType = "class_label",
         data_model_grapher: Optional[DataModelGraph] = None,
         data_model_graph_explorer: Optional[DataModelGraphExplorer] = None,
         parsed_data_model: Optional[dict] = None,
+        graph_data_model: Optional[nx.MultiDiGraph] = None,
+        data_model_graph_pickle: Optional[str] = None,
     ) -> None:
         self.path_to_jsonld = path_to_jsonld
 
         self.jsonld = load_json(self.path_to_jsonld)
+        if graph_data_model is not None:
+            self.graph_data_model = graph_data_model
+        elif data_model_graph_pickle is not None:
+            data_model_graph = read_pickle(data_model_graph_pickle)
+            if not isinstance(data_model_graph, nx.MultiDiGraph):
+                raise ValueError(
+                    "The data model graph must be a networkx MultiDiGraph object."
+                )
+            self.graph_data_model = data_model_graph
 
         # Parse Model
-        if not parsed_data_model:
+        if parsed_data_model is None:
             data_model_parser = DataModelParser(
                 path_to_data_model=self.path_to_jsonld,
             )
             parsed_data_model = data_model_parser.parse_model()
 
         # Instantiate DataModelGraph
-        if not data_model_grapher:
+        if data_model_grapher is None:
             data_model_grapher = DataModelGraph(parsed_data_model, data_model_labels)
-
-        # Generate graph
-        self.graph_data_model = data_model_grapher.graph
+            # Generate graph
+            self.graph_data_model = data_model_grapher.graph
```

Comment:
Will self.graph_data_model always be assigned a value? The logic here is kind of confusion, and it's nto clear it always will be.